RecoTracker/TkTrackingRegions/plugins/CandidatePointSeededTrackingRegionsProducer.h

```diff
@@ -0,0 +1,325 @@
+            GlobalPoint( v->x(), v->y(), v->z() ),
+            (m_mode == VERTICES_FIXED) ? m_zErrorVetex : m_nSigmaZVertex*v->zError()
+        ));
+        ++n_vert;
+      }
+      // no-vertex fall-back case:
+      if (origins.empty())
+      {
+        origins.push_back( std::make_pair(
+            default_origin,
+            (m_nSigmaZBeamSpot > 0.) ? m_nSigmaZBeamSpot*bs->z0Error() : m_zErrorBeamSpot
+        ));
+      }
+    }
+    
+    const MeasurementTrackerEvent *measurementTracker = nullptr;
+    if(!token_measurementTracker.isUninitialized()) {
+      edm::Handle<MeasurementTrackerEvent> hmte;
+      e.getByToken(token_measurementTracker, hmte);
+      measurementTracker = hmte.product();
+    }
+
+    // create tracking regions (maximum MaxNRegions of them) in directions of the
+    // objects of interest (we expect that the collection was sorted in decreasing pt order)
+    int n_regions = 0;
+    size_t n_points = etaPoints.size();
+
+    for(size_t i = 0; i < n_objects && n_regions < m_maxNRegions; ++i )
+    {
+      const reco::Candidate & object = (*objects)[i];
+      double eta_Cand = object.eta();
+      double phi_Cand = object.phi();
+
+      for(size_t j = 0 ; j < n_points && n_regions < m_maxNRegions; ++j)
+      {
+
+	double eta_Point = etaPoints[j];
+	double phi_Point = phiPoints[j];
+
+	double dEta_Cand_Point = fabs(eta_Cand-eta_Point);
+	double dPhi_Cand_Point = fabs(deltaPhi(phi_Cand,phi_Point));
+
+	//Check if there is an overlap between Candidate- and Point-based regions of interest
+	if(dEta_Cand_Point > (m_deltaEta_Cand + m_deltaEta_Point) || dPhi_Cand_Point > (m_deltaPhi_Cand + m_deltaPhi_Point)) continue;
+
+	//Determines boundaries of intersection of RoIs
+	double etaMin_RoI = std::max(eta_Cand-m_deltaEta_Cand,eta_Point-m_deltaEta_Point);
+	double etaMax_RoI = std::min(eta_Cand+m_deltaEta_Cand,eta_Point+m_deltaEta_Point);
+
+	double phi_Cand_minus  = normalizedPhi(phi_Cand-m_deltaPhi_Cand);
+	double phi_Point_minus = normalizedPhi(phi_Point-m_deltaPhi_Point);
+	double phi_Cand_plus  = normalizedPhi(phi_Cand+m_deltaPhi_Cand);
+	double phi_Point_plus = normalizedPhi(phi_Point+m_deltaPhi_Point);
+
+	double phiMin_RoI = deltaPhi(phi_Cand_minus,phi_Point_minus)>0. ? phi_Cand_minus : phi_Point_minus ;	
+	double phiMax_RoI = deltaPhi(phi_Cand_plus,phi_Point_plus)<0. ? phi_Cand_plus : phi_Point_plus;
+
+	//Determines position and width of new RoI
+	double eta_RoI = 0.5*(etaMax_RoI+etaMin_RoI);
+	double deltaEta_RoI = etaMax_RoI - eta_RoI;
+
+	double phi_RoI = 0.5*(phiMax_RoI+phiMin_RoI);
+	if( phiMax_RoI < phiMin_RoI ) phi_RoI-=M_PI;
+	phi_RoI = normalizedPhi(phi_RoI);
+	double deltaPhi_RoI = deltaPhi(phiMax_RoI,phi_RoI);
+
+       	double x = std::cos(phi_RoI);
+	double y = std::sin(phi_RoI);
+	double theta = 2*std::atan(std::exp(-eta_RoI));
+	double z = 1./std::tan(theta);
+
+	GlobalVector direction( x,y,z );
+      
+	for (size_t  j=0; j<origins.size() && n_regions < m_maxNRegions; ++j)
```

Comment:
Loop variable better be `k` or something else to not clash with `j` or `i` of the inner loops (or convert to range-for).

Reply:
Even if in practice this appears to be safe (the `j` in inner block refers to the `j` declared in the inner block), but to not confuse the reader.

Reply:
Definitely makes sense
Thanks for spotting this